core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -4017,6 +4036,12 @@ object ZIO extends ZIOCompanionPlatformSpecific {
   def untraced[R, E, A](zio: ZIO[R, E, A]): ZIO[R, E, A] =
     zio.untraced
 
+  /**
+   * Updates a state in the environment with the specified function.
+   */
+  def updateState[S: Tag](f: S => S): ZIO[Has[State[S]], Nothing, Unit] =
```

Comment:
Could potentially add more variants here consistent with the ones on `Ref` but not sure how useful they would be given that changes are occurring on single fiber and would also like to keep the number of ways to do things down.

Reply:
Exactly. Best to keep it minimal.